web/admin: revamped rbac and user settings tabs

BeryJu opened the pull request:
<!--
👋 Hi there! Welcome.

Please check the Contributing guidelines: https://goauthentik.io/developer-docs/#how-can-i-contribute
-->

## Details

A whole load of minor-ish fixes

- rename role permission tables to match user tables
- use separate tabs
- rework role page to match user page
- show all of users permission tabs on one main tab
- move stuff into vertical subtab
- fix some leftover html?
- switch non-embedded permission page to use vertical tabs
- web/admin: fix duplicate RBAC preview banner on permission modal

closes #7675
closes #6760

---

## Checklist

-   [ ] Local tests pass (`ak test authentik/`)
-   [ ] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [ ] The code has been formatted (`make web`)
-   [ ] The translation files have been updated (`make i18n-extract`)

If applicable

-   [ ] The documentation has been updated
-   [ ] The documentation has been formatted (`make website`)

BeryJu committed:
web/admin: fix duplicate RBAC preview banner on permission modal

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
switch non-embedded permission page to use vertical tabs

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
fix some leftover html?

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
move stuff into vertical subtab

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
show all of users permission tabs on one main tab

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
rework role page to match user page

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
use separate tabs

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
rename role permission tables to match user tables

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
rename to credentials and tokens

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
add country icon to session list

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
add oauth access token list

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
add helper to get relative time

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
use pfdivider

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
replace plain hr with pf-c-divider

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
use new logic for showing relative time in charts

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
use consistent relative time for event display

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
remove more leftovers

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
fix some alignment issues on the admin dashboard

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
update storybook map

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
add sanity check to event app lookup

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
make api drawer header fixed

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
fix table padding for toggle

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
fix notification drawer for user interface

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
enable system task search

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
fix formatting, exclude generated script from formatting

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

kensternberg-authentik commented on web/src/admin/events/EventListPage.ts:
```diff
@@ -82,9 +83,9 @@ export class EventListPage extends TablePage<Event> {
                 <small>${item.app}</small>`,
             EventUser(item),
-            html`<span>${item.created?.toLocaleString()}</span>`,
+            html`<div>${getRelativeTime(item.created)}</div>
```
Nifty.  I tend to reverse these: "Dec 10, 2023 (23 days ago)", but I think the human brain appreciates having a "distance from here / distance from now" along with a location or date.

tanberry replied:
My brain is quite happy to get that relative info.

kensternberg-authentik committed:
web: minor fixes

There's a renderer (it's not a component, not yet) for producing definition lists without
the risk of missing a class or tag.

Breaking conditionally rendered components out to make their use easier to identify.

kensternberg-authentik commented on web/src/admin/roles/RoleViewPage.ts:
```diff
@@ -83,73 +81,61 @@ export class RoleViewPage extends AKElement {
+                            ${renderDescriptionList([
+                                [msg("Name"), this._role.name],
+                                [msg("Edit"), this.renderUpdateControl(this._role)],
+                            ])}
```
Nice, small description list renderer.  :-)

BeryJu replied:
can we turn this into a component? Currently when using this we'll also have to manually import the descriptor CSS

kensternberg-authentik replied:
We can, but you already have in existing cases, so I went with a lazier solution.  I do intend to create a `ak-description-list` component someday.  This is a compromise; I think it's a good one that doesn't introduce any new semantics or CSS shadow boundary issues, and it puts markers in the ground for where to target our description lists in the future.

kensternberg-authentik commented on web/src/elements/rbac/ObjectPermissionsPage.ts:
```diff
@@ -76,4 +86,78 @@ export class ObjectPermissionPage extends AKElement {
             </ak-tabs>`;
     }
+
+    renderCoreUser() {
```
@BeryJu broke out `renderControlButtons` above in the UserView page, and it's a good practice; break out big pieces of functionality, especially when they're rendered conditionally, and name them.

kensternberg-authentik replied:
Two reasons, actually: the first is that [Typescript generally recommends against it](https://www.typescriptlang.org/docs/handbook/2/functions.html), as inference of return type is very good and the only reason to do it is if you don't trust the function body itself to generate the correct return type, and the other is that Lit's re-render algorithm is smarter with `nothing` rather than `html\`\`` , but doesn't conform to TemplateResult without a lot of fussiness.

kensternberg-authentik reviewed (approved):
I approve your changes, but I made two changes myself, so I'll need your assent.  :-)

BeryJu commented on web/src/admin/roles/RoleViewPage.ts:
```diff
@@ -71,7 +72,7 @@ export class RoleViewPage extends AKElement {
     }
 
-    render(): TemplateResult {
+    render() {
```
any reason for removing the type annotations?

kensternberg-authentik replied:
Two reasons, actually: the first is that [Typescript generally recommends against it](https://www.typescriptlang.org/docs/handbook/2/functions.html), as inference of return type is very good and the only reason to do it is if you don't trust the function body itself to generate the correct return type, and the other is that Lit's re-render algorithm is smarter with `nothing` rather than `html\`\`` , but doesn't conform to TemplateResult without a lot of fussiness.

BeryJu committed:
fix prettier

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
fix outpost form

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
fix more flaky tests

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
re-create locale

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
add some description for different permission views

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
fix system task search

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
update docs

Signed-off-by: Jens Langhammer <jens@goauthentik.io>
